src/TemplateData.js

```diff
@@ -248,14 +248,37 @@ class TemplateData {
     return globalData;
   }
 
+  async getInitialGlobalData() {
+    let globalData = {};
+    if (this.config.globalData) {
+      let keys = Object.keys(this.config.globalData);
+      for (let j = 0; j < keys.length; j++) {
+        let returnValue = this.config.globalData[keys[j]];
+
+        if (typeof returnValue === "function") {
+          returnValue = await returnValue();
+        }
+        globalData[keys[j]] = returnValue;
+      }
+    }
+    return globalData;
+  }
+
   async getData() {
     let rawImports = this.getRawImports();
 
+    let initialGlobalData = await this.getInitialGlobalData();
+
     if (!this.globalData) {
       let globalJson = await this.getAllGlobalData();
 
       // OK: Shallow merge when combining rawImports (pkg) with global data files
-      this.globalData = Object.assign({}, globalJson, rawImports);
+      this.globalData = Object.assign(
```

Comment:
Ah, sorry a few more thoughts: Not saying it does—just trying to think through it—but I wonder if this might need to work with deep merging? I think previously global data was just a shallow copy which has been fine because it’s first in the chain but we might get into some tricky situations. (After typing this all out—I think it’s probably fine to punt on this one for now but I just wanted to log it)

Related: As it stands, it looks like a global data file would overwrite the same key entry set in `addGlobalData`. Is this what we want or the other way around? I think `addGlobalData` will probably be plugin-land so I think global data files should probably have precedence. Okay, I’m done now—looks good!